cmd/app.go

```diff
@@ -238,30 +260,93 @@ func appLs(ctx *cli.Context) error {
 	}
 
 	collection, err := c.ProjectClient.App.List(defaultListOpts(ctx))
+	if err != nil {
+		return err
+	}
+
 	writer := NewTableWriter([][]string{
 		{"ID", "ID"},
 		{"NAME", "App.Name"},
 		{"STATE", "App.State"},
+		{"CATALOG", "Catalog"},
+		{"TEMPLATE", "Template"},
 		{"VERSION", "Version"},
 	}, ctx)
 
 	defer writer.Close()
 
 	for _, item := range collection.Data {
-		parsedExternal, err := parseExternalID(item.ExternalID)
+		appData := &AppData{
+			ID:  item.ID,
+			App: item,
+		}
+
+		appExternalID := item.ExternalID
+		var appTemplateFiles map[string]string
+		if appExternalID == "" {
+			// add namespace prefix to AppRevisionID to create a Rancher API style ID
+			appRevisionID := strings.Replace(item.ID, item.Name, item.AppRevisionID, -1)
+
+			appRevision, err := c.ProjectClient.AppRevision.ByID(appRevisionID)
+			if err != nil {
+				return err
+			}
+
+			appTemplateFiles = appRevision.Status.Files
+		}
+
+		appData.Catalog, appData.Template, appData.Version, err = parseTemplateInfo(appExternalID, appTemplateFiles)
 		if err != nil {
 			return err
 		}
-		writer.Write(&AppData{
-			ID:      item.ID,
-			App:     item,
-			Version: parsedExternal["version"],
-		})
+
+		writer.Write(appData)
 	}
 	return writer.Err()
 
 }
 
+func parseTemplateInfo(appExternalID string, appTemplateFiles map[string]string) (templateCatalog, templateName, templateVersion string, err error) {
+	if appExternalID != "" {
+		parsedExternal, parseErr := parseExternalID(appExternalID)
+		if parseErr != nil {
+			err = errors.Wrap(parseErr, "can't parse ExternalID from app")
+			return
+		}
+
+		templateCatalog = parsedExternal["catalog"]
+		templateName = parsedExternal["template"]
+		templateVersion = parsedExternal["version"]
+
+		return
+	}
+
+	for fileName, fileContent := range appTemplateFiles {
+		if strings.HasSuffix(fileName, "/Chart.yaml") || strings.HasSuffix(fileName, "/Chart.yml") {
+			content, decodeErr := base64.StdEncoding.DecodeString(fileContent)
+			if decodeErr != nil {
+				err = errors.Wrap(decodeErr, "can't decode Chart.yaml from app")
+				return
+			}
+
+			version := &chartVersion{}
+			unmarshalErr := yaml.Unmarshal(content, version)
+			if unmarshalErr != nil {
+				err = errors.Wrap(unmarshalErr, "can't parse Chart.yaml from app")
+				return
+			}
+
+			templateCatalog = "local folder"
```

Comment:
nit: s/folder/directory - folder tends to mean more of a UI construct and a windows thing.